x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/correlation/CountCorrelationFunction.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        indicator.writeTo(out);
+    }
+
+    @Override
+    public String getName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public int hashCode() {
+        return NAME.getPreferredName().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+        CountCorrelationFunction other = (CountCorrelationFunction) obj;
+        return Objects.equals(indicator, other.indicator);
+    }
+
+    @Override
+    public double execute(CorrelativeValue y) {
+        if (indicator.getExpectations().length != y.getExpectations().length) {
+            throw new AggregationExecutionException(
+                "value lengths do not match; indicator.expectations ["
+                    + indicator.getExpectations().length
+                    + "] and number of buckets ["
+                    + y.getExpectations().length
+                    + "]. Unable to calculate correlation"
+            );
+        }
+        final double xMean;
+        final double xVar;
+        if (indicator.getFractions() == null) {
+            double sum = CorrelationFunction.sum(indicator.getExpectations());
+            xMean = sum / indicator.getExpectations().length;
+            double var = 0;
+            for (double v : indicator.getExpectations()) {
+                var += Math.pow(v - xMean, 2);
+            }
+            xVar = var / indicator.getExpectations().length;
+        } else {
+            double mean = 0;
+            for (int i = 0; i < indicator.getExpectations().length; i++) {
+                mean += indicator.getExpectations()[i] * indicator.getFractions()[i];
+            }
+            xMean = mean;
+            double var = 0;
+            for (int i = 0; i < indicator.getExpectations().length; i++) {
+                var += Math.pow(indicator.getExpectations()[i] - xMean, 2) * indicator.getFractions()[i];
+            }
+            xVar = var;
+        }
+        final double weight = CorrelationFunction.sum(y.getExpectations())/indicator.getDocCount();
+        if (weight > 1.0) {
+            throw new AggregationExecutionException(
+                "doc_count of indicator must be larger than the total count of the correlating values indicator count ["
+                    + indicator.getDocCount()
+                    + "] correlating value total count ["
+                    + CorrelationFunction.sum(y.getExpectations())
+                    + "]"
+            );
+        }
+        final double yMean = weight;
+        final double yVar = (1 - weight) * yMean * yMean + weight * (1 - yMean) * (1 - yMean);
+        double xyCov = 0;
+        if (indicator.getFractions() != null) {
```

Comment:
nit comment but I think it'd make it easier to read this if you invert the if logic here (or the one above) so that in both cases we're dealing first with the same case (whether it is there are fractions or not).